### PR TITLE
[WIP]Remove jquery-ui from batch 5 of scripts

### LINKED ISSUE #3010: Remove jquery-ui from batch 5 of scripts
Remove jquery-ui from following scripts:
interface/usergroup/facility_user.php
interface/usergroup/facility_user_admin.php
interface/usergroup/usergroup_admin.php
interface/weno/ajax_sample_code.php
interface/weno/confirm.php
library/custom_template/custom_template.php
library/custom_template/personalize.php

### PR DESCRIPTION
**Pull request by im-Amitto**
Fixes #3010

### CODE REVIEW

File: interface/weno/ajax_sample_code.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
 <html>
 <head>
 <title><?php print xlt("Weno Sample Code") ?></title>
-<?php Header::setupHeader(['jquery-ui', 'jquery-ui-sunny']); ?>
+<?php Header::setupHeader([]); ?>
```
Comment (tywrenn) [inline interface/weno/ajax_sample_code.php]: Remove [] brackets

File: interface/weno/confirm.php [hunk 2]
```diff
@@ -34,7 +34,7 @@
 
 <html>
 <head>
-    <?php Header::setupHeader(['jquery-ui', 'jquery-ui-sunny']); ?>
+    <?php Header::setupHeader([]); ?>
```
Comment (tywrenn) [inline interface/weno/confirm.php]: Remove [] brackets

Comment (tywrenn) [inline interface/weno/confirm.php]: Remove [] brackets

File: interface/weno/ajax_sample_code.php (hunk 1 above)
Comment (tywrenn) [inline interface/weno/ajax_sample_code.php]: Remove [] brackets

**Comment by tywrenn** [comment]
Done with my review here

File: library/custom_template/custom_template.php [hunk 3]
```diff
@@ -177,9 +177,9 @@ function refreshme() {
                                 <?php if (!$isNN) { ?>
                                     <td>
                                         <div id="searchCriteria">
-                                            <div class="select-box form-inline" style="margin-bottom:5px;">
+                                            <div class="select-box form-inline mb-5">
```
Comment (tywrenn) [inline library/custom_template/custom_template.php]: This should be `mb-1`

File: library/custom_template/personalize.php [hunk 4]
```diff
@@ -451,7 +451,7 @@ function personalize_save() {
         </tr>
         <tr class="text">
             <td colspan="3">
-                <div style="width:100%;overflow:auto;height:150px" id="itemdiv"></div>
+                <div class="w-100" style="overflow:auto;height:150px" id="itemdiv"></div>
```
Comment (tywrenn) [inline library/custom_template/personalize.php]: `overflow:auto` can be replaced with the `overflow-auto` class in BS4

**Comment by tywrenn** [comment]
@im-Amitto Please fix Travis but other than that @bradymiller please test to see if these files function normally

File: library/custom_template/custom_template.php [hunk 5]
```diff
@@ -50,7 +50,7 @@ function listitemCode($strDisp, $strInsert)
 ?>
 <html>
```
Comment (bradymiller) [inline library/custom_template/custom_template.php]: This is showing ' TypeError: $(...).sortable is not a function'

image

(I'll show a video later on how to use Nation Note, so you can recreate/fix this bug)

**Comment by bradymiller** [comment]
Here's video showing use of Nation Notes:
https://www.loom.com/share/28408982701e4163b8ce6b4b6f17452b

**Comment by im-Amitto** [comment]
@bradymiller how can i add a few items in their?
Screenshot from 2020-03-01 18-05-30

**Comment by bradymiller** [comment]
hi @im-Amitto , When you click on Patient Details, by default, it should show the following:
image

**Comment by im-Amitto** [comment]
@bradymiller that functionality works without jquery and the function which is causing error is linked to the `#menu5 div` which is shown in previous picture i uploaded. It's empty and i am not sure how can i put some data in there to test the functionality.

**Comment by bradymiller** [comment]
hi @im-Amitto , Here's how to add to the that div:

https://www.loom.com/share/340d68d8471d4648b3de262c763162a4

File: library/custom_template/custom_template.php (hunk 5 above)
Comment (im-Amitto) [reply library/custom_template/custom_template.php]: can't believe i am doing this for two days and have't found a optimal solution :cry:

Comment (bradymiller) [reply library/custom_template/custom_template.php]: we may need to call in super Jerry (aka @sjpadgett ) :)

tenor

Comment (im-Amitto) [reply library/custom_template/custom_template.php]: @bradymiller i did created a workaround using interact.js but not getting the real feel.

Comment (sjpadgett) [reply library/custom_template/custom_template.php]: > I can't replace the jquery-ui function till i don't what is expected of him.

@im-Amitto 
There will be several places in codebase where jquery-ui was included to keep dialog from having to include. This is one of them i'm pretty sure.

**Comment by sjpadgett** [comment]
@im-Amitto , where are you on this, how can I help?
Are you still needing drag/drop. Did you get ckeditor to load it's config(looks like)?

**Comment by im-Amitto** [comment]
@sjpadgett I just pushed the solution that i created to replace sortable using drag/drop but i am not sure about how good it is.
CC - @bradymiller

File: library/custom_template/custom_template.php [hunk 6]
```diff
@@ -141,14 +141,115 @@ function refreshme() {
...
+            var x = (parseFloat(target.getAttribute('data-x')) || 0) + event.dx
+            var y = (parseFloat(target.getAttribute('data-y')) || 0) + event.dy
+            target.style.webkitTransform =
+                target.style.transform =
+                'translate(' + x + 'px, ' + y + 'px)'
+            target.setAttribute('data-x', x)
+            target.setAttribute('data-y', y)
+        }
+        function init() {
+            interact('.droppable').dropzone({
```
Comment (sjpadgett) [inline library/custom_template/custom_template.php]: Lets see if we can add to utility.js as a callable for other use case in code and convenience.
maybe oeSortable(....)

Nice job on the JS

Comment (im-Amitto) [reply library/custom_template/custom_template.php]: I have moved it to utility and also made a few that i think would be nice but may not be desirable .Alos, i have added a callback parameter to the sortable function so we can link an event(if we want) required to take place whenver list changes.

**Comment by mdsupport** [comment]
@im-Amitto 
Hopefully you have looked around for jqui sortable alternatives like [this discussion](https://github.com/solidusio/solidus/issues/2469).  To play with that part of code, you need to use the buttons at bottom which let you setup groups within a top group (soap).  Once you select from available groups, you can add sentences in a tiny textbox.
Drag/Drop/Sort functions are to allow users to rearrange those sentences.

@sjpadgett has a much nicer way to enter and manage superset of these templates.  So unless there is a ready solution floating there, maybe project can temporarily loose sort function and follow up your PR with another that adds a link to better interface.  This component needs major update since CKEditor has since incorporated inline html editor.

**Comment by bradymiller** [comment]
@im-Amitto , Testing well. Nice js and I think is good enough to bring into codebase as long as @sjpadgett agrees.

**Comment by im-Amitto** [comment]
> @im-Amitto
> Hopefully you have looked around for jqui sortable alternatives like [this discussion](https://github.com/solidusio/solidus/issues/2469). 

Yeah, i have gone through multiple implementation but i am trying to recreate the functionality with `interact.js` since we are already using it in codebase.

**Comment by sjpadgett** [comment]
@mdsupport Both good recommends however, how would it look if I preach not taking away features then turn around and endorse removing one:) This would be one I could pretend to not notice though:)
But, i'd still worry long time users would be upset. NN, like Layout, is very beloved my many users so, for now, this will get it done. We really should be looking at A.I for templating. Now that'd be fun project for a young, eagar engineer! If one would want Google to take notice for project then, A.I or FHIR would almost ensure acceptance.

I wouldn't push back to much on a new library but then again, how soon would we be back here again when that library goes quite? This is one of the reasons i've been trying to develop our own set of utilities where possible even though it needs to be restructured some to act more like a JS library.

Agree that having context and category selections available in NN just seems to make sense to me. It would not be hard to do with the call from NN being the default Context. Especially with updating ckeditor and the new goodies that would provide.

As always, appreciate the feedback. Your sortable discussion link was a nice ref. May help with datepicker.
@im-Amitto i'll review shortly as I have a few concerns. btw: doesn't anyone sleep anymore:)

**Comment by bradymiller** [comment]
@sjpadgett , Of course we sleep:
tenor-2

**Comment by im-Amitto** [comment]
A picture is a thousand words then a video meant a million words isn't it :)

**Comment by sjpadgett** [comment]
@im-Amitto A couple issues.
- drag cursor should only appear from category template top or change when hover over edit
- left click to drag will select other template text ie
image

If you have to bring in a library to handle this or keep going. up to you.
Here is an idea of a production progress template
image

**Comment by im-Amitto** [comment]
@sjpadgett I was thinking of doing it in my own but i think importing a library will be better as we will be needing it at many places.

**Comment by mdsupport** [comment]
@im-Amitto When migrating from a package, it is best to create a drop-in replacement that keeps getting richer as you go thru testing and discover features used in package at hand.

Re. sortable, there is 1:1 highly specific replacement in [html5sortable](https://github.com/lukasoppermann/html5sortable) with dependency on html5 and non-IE11 browsers.

I would recommend 
1. Create a pure .js script (e.g. oeUI_1.js) which will contain commonly used jqUI feature replacement.
2. If you need php/server input, make server script build that object rather than .js.php approach which prevents caching.
3. Let that library do client stuff for all elements with specific class (e.g. `oe-sortable`).
4. Let the .js be the bridge that accepts links for ajax processors as specified in php script and handle specific events.
5. Scan entire codebase for `.sortable` and try the approach on at least 3 to cover variations.

Best.

**Comment by im-Amitto** [comment]
@mdsupport Sounds like a good plan :)

**Comment by sjpadgett** [comment]
@im-Amitto Even if you can/need find replacement for interact.js if provides same capabilities. I'll do change over if you decide a better library exist.

**Comment by mdsupport** [comment]
@im-Amitto, @sjpadgett - Basic suggestion is to move any and all feature specific js code from php scripts to an common script.  Actual work can be done by interact or by any other components.  This did not happen with jqui otherwise this would have been a simple(r) PR.

**Comment by tywrenn** [comment]
@im-Amitto Still working on this?

**Comment by im-Amitto** [comment]
@tywrenn I actually shifted my focus on flutter openemr flutter but I will try to fix it in within 48 hours if it doesn't work let's close it as I can't abandon the flutter repo for too long.

**Comment by im-Amitto** [comment]
I mistakenly deleted the forked repo so closing this PR

**Pull request closed without merging**